subprojects/execution/build.gradle.kts

```diff
@@ -26,10 +26,11 @@ dependencies {
     implementation(project(":files"))
     implementation(project(":snapshots"))
     implementation(project(":model-core"))
-    implementation(project(":persistent-cache"))
     implementation(project(":build-cache"))
     implementation(project(":build-cache-packaging"))
 
+    api(project(":persistent-cache"))
+
```

Comment:
`api` dependencies typically go above `implementation` ones.

But how come this is the first `api` dependency? We don't seem to need `files` for example. Shouldn't that be an `api` dependency? And some of the others seem like candidates, too...

Reply:
I agree that some other dependencies also probably should be API dependencies. It looks like this hasn't been a problem up to now, i.e we have been adding the extra dependencies in the consumers manually. Do we want to continue doing that and move the dependency on `persistent-cache` back to implementation?

Reply:
I'd rather follow what we did before in this PR, and discuss with @jjohannes how we wanted to approach this on a more general level.